beets/art.py

```diff
@@ -190,19 +190,26 @@ def check_art_similarity(log, item, imagepath, compare_threshold):
 
 
 def extract(log, outpath, item):
+    """Extract art from item into outpath with an appropriate file extension,
+    or into a tmp file if outpath is unset.
+    """
     art = get_art(log, item)
-    outpath = bytestring_path(outpath)
     if not art:
         log.info(u'No album art present in {0}, skipping.', item)
         return
 
-    # Add an extension to the filename.
+    # Discover image file extension
     ext = mediafile.image_extension(art)
     if not ext:
         log.warning(u'Unknown image type in {0}.',
                     displayable_path(item.path))
         return
-    outpath += bytestring_path('.' + ext)
+
+    # Append extension to outpath / create tmp file with extension
```

Comment:
```suggestion
    # Append the extension to destination path, creating a temporary file
    # if no destination is specified.
```

We generally like to use complete sentences, with periods, in comments.